Projektarbeit III/Assets/Scripts/Player/PlayerController/MovementUtils.cs

```diff
@@ -0,0 +1,21 @@
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public static class MovementUtils
+{
+    public static void ApplyHorizontalMovement(Rigidbody2D rb, InputAction movementAction, float moveSpeed)
+    {
+        float moveInput = movementAction.ReadValue<Vector2>().x;
+
+        if (Mathf.Abs(rb.linearVelocity.x) > moveSpeed)
+        {
+            // Maintain the current horizontal velocity to preserve momentum
+            rb.linearVelocity = new Vector2(rb.linearVelocity.x + moveInput * moveSpeed * Time.deltaTime, rb.linearVelocity.y);
```

Comment:
Why? xD Isn't
```suggestion
            rb.linearVelocityX += moveInput * moveSpeed * Time.deltaTime;
```
doing exactly the same?